### ISSUE TITLE
can add a mape objective function in lightgbm?

### ISSUE THREAD

**Issue by hlee13** [Microsoft/LightGBM#624]
Such as the title.
reference to：https://en.wikipedia.org/wiki/Mean_absolute_percentage_error

could show the mape objective function code segment in python now.

**Comment by wxchan** [comment]
check https://github.com/dmlc/xgboost/issues/1818

**Comment by Laurae2** [comment]
@hlee13 You can contribute by adding MAPE to LightGBM in C++, or you can use a custom metric in R/Python.

**Comment by guolinke** [comment]
added in #1199
